front/migrations/20241211_parents_migrator.ts

```diff
@@ -89,7 +114,45 @@ const migrators: Record<ConnectorProvider, ProviderMigrator | null> = {
   snowflake: null,
   webcrawler: null,
   zendesk: null,
-  confluence: null,
+  confluence: {
+    transformer: (nodeId, parents) => {
+      // case where we got old IDs exclusively: we add the new IDs to them
+      if (
+        parents.every(
+          (parent) =>
+            parent.startsWith(ConfluenceOldIdPrefix.Page) ||
+            parent.startsWith(ConfluenceOldIdPrefix.Space)
+        )
+      ) {
+        return [...parents, ...parents.map(convertConfluenceOldIdToNewId)];
+      }
+      // checking that we got a mix of old and new IDs, with the old ones matching the new ones
+      for (const parent of parents) {
```

Comment:
you're just checking the validity of the code above here no?

Reply:
i'm checking the validity of the new code in connectors that upserts the parents